### PR TITLE
Detect Threat indicators for VMware ESXi servers

### PR DESCRIPTION
**Pull request by shashank-elastic**
## Issues
#2694
#2695
#2696
#2697
#2698
#2699

## Summary

**_ESXI Discovery via Grep:_** Identifies instances where a process named 'grep', 'egrep', or 'pgrep' is started on a Linux system with arguments related to virtual machine (VM) files, such as "vmdk", "vmx", "vmxf", "vmsd", "vmsn", "vswp", "vmss", "nvram", or "vmem".

**_ESXI Discovery via Find:_** Identifies instances where the 'find' command is started on a Linux system with arguments targeting specific VM-related paths, such as "/etc/vmware/", "/usr/lib/vmware/", or "/vmfs/*".

 **_ESXI Timestomping using Touch Command:_** Identifies instances where the 'touch' command is executed on a Linux system with the "-r" flag, which is used to modify the timestamp of a file based on another file's timestamp. The rule targets specific VM-related paths, such as "/etc/vmware/", "/usr/lib/vmware/", or "/vmfs/*"

**_Suspicious Termination of ESXI Process:_** Identifies instances where VMware processes, such as "vmware-vmx" or "vmx," are terminated on a Linux system by a "kill" command. The rule monitors for the "end" event type, which signifies the termination of a process.

**_Suspicious Renaming of ESXI Files:_** Identifies instances where VMware-related files, such as those with extensions like ".vmdk", ".vmx", ".vmxf", ".vmsd", ".vmsn", ".vswp", ".vmss", ".nvram", and ".vmem", are renamed on a Linux system. 

**_Suspicious Renaming of ESXI index.html File:_** Identifies instances where the "index.html" file within the "/usr/lib/vmware/*" directory is renamed on a Linux system.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/linux/defense_evasion_esxi_suspicious_timestomp_touch.toml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+]
+risk_score = 47
+rule_id = "30bfddd7-2954-4c9d-bbc6-19a99ca47e23"
+severity = "medium"
+tags = ["Elastic", "Host", "Linux", "Threat Detection", "Defense Evasion"]
+timestamp_override = "event.ingested"
+type = "eql"
+query = '''
+process where host.os.type == "linux" and event.type == "start" and process.name : "find" and
+process.args : ("/etc/vmware/*", "/usr/lib/vmware/*", "/vmfs/*")
```
Comment (imays11) [inline rules/linux/defense_evasion_esxi_suspicious_timestomp_touch.toml:34]: I think you accidentally put the wrong query for this one

File: rules/linux/defense_evasion_rename_esxi_index_file.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.5.0"
+updated_date = "2023/04/11"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies instances where the "index.html" file within the "/usr/lib/vmware/*" directory is renamed on a Linux system.
+The rule monitors for the "rename" event action associated with this specific file and path,
+which could indicate malicious activity..
```
Comment (imays11) [inline rules/linux/defense_evasion_rename_esxi_index_file.toml:14]:
```suggestion
which could indicate malicious activity.
```

**Review by imays11 — COMMENTED** [review]
Cool set of rules. Just one fix for the accidental query mix-up, but I saw the correct one in the rule issue and it looks good

File: rules/linux/defense_evasion_esxi_suspicious_timestomp_touch.toml (hunk 1 above)
Comment (shashank-elastic) [reply rules/linux/defense_evasion_esxi_suspicious_timestomp_touch.toml]: Have updated the Query.
Also have double checked all my query findings!

**Pull request merged**